src/sagemaker/workflow/emr_step.py

```diff
@@ -60,6 +63,20 @@ def to_request(self) -> RequestType:
 
         return config
 
+    def validate_prop_dicts(self):
+        """Validate k,v structure of properties dictionaries, err if not provided"""
+        if isinstance(self.properties, list) and self.properties:
```

Comment:
Nit: we don't need the `and self.properties` as 
- If self.properties is None, the isinstance will return False
- If self.properties is empty, it would not go into the for loop